hr_expense_invoice/models/hr_expense.py

```diff
@@ -21,6 +23,37 @@ class HrExpense(models.Model):
         copy=False,
     )
 
+    def action_expense_create_invoice(self):
+        invoice_lines = [
+            (
+                0,
+                0,
+                {
+                    "product_id": self.product_id.id,
+                    "name": self.name,
+                    "price_unit": self.unit_amount,
+                    "quantity": self.quantity,
+                    "account_id": self.account_id.id,
+                    "analytic_account_id": self.analytic_account_id.id,
+                    "tax_ids": [(6, 0, self.tax_ids.ids)],
+                },
+            )
+        ]
+        invoice = (
+            self.env["account.move"]
+            .with_context(default_type="in_invoice")
+            .create(
+                {
+                    "type": "in_invoice",
```

Comment:
Why do you need to specify the type twice? I'd say the correct one would be through the context.